### PR TITLE
add optional parameter keyevent that defines the type of the keyevent

### PR DESCRIPTION
**Pull request by gurschitz**
I've implemented what we've discussed in #2 by adding an optional parameter keyevent that can be passed within the object in the second argument. If it's not passed, then the keyevent defaults to `keydown`. Other possibilities are `keyup` and `keypress`. I've written tests to confirm this behaviour.

I have also refactored the error handling logic by using `invariant`. It also checks if the keyevent is valid (i.e. if it's one of `keydown`, `keyup` or `keypress`).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
 
   if (!Array.isArray(detectKeys)) {
     allowedKeys = [];
     // eslint-disable-next-line no-console
     console.warn('Keys should be array!');
   }
+
   allowedKeys = convertToAsciiEquivalent(allowedKeys);
 
   const handleKeydown = event => {
```
Comment (haldarmahesh) [inline src/index.js]: Now that this function is going to be common for all 3 events, let's name it better.
`handleEvent` or some other meaningful name.
WYT?

File: src/index.js [hunk 2]
```diff
@@ -1,21 +1,27 @@
 const { useEffect } = require('react');
+const invariant = require('invariant');
 const { onKeyPress, convertToAsciiEquivalent, getAsciiCode } = require('./keys.js');
 
-const useKey = (callback, { detectKeys = [] } = {}, { dependencies = [] } = {}) => {
-  let allowedKeys = detectKeys;
-  if (!window || !window.document || !callback) {
-    throw new Error();
-  }
+const VALID_KEYEVENTS = ['keydown', 'keyup', 'keypress'];
```
Comment (haldarmahesh) [inline src/index.js]: Please rename to `VALID_KEY_EVENTS` 
That missing `_`

File: src/__test__/index.test.js [hunk 3]
```diff
@@ -6,33 +6,36 @@ import { render, flushEffects, cleanup, fireEvent } from 'react-testing-library'
...
-      render(<TestComponent detectKeys={[12]} />);
-    } catch (e) {
-      // success
-    }
+    expect(() => render(<TestComponent detectKeys={[12]} />)).toThrowError();
   });
 
-  test('when keys is not array it passes with warns', () => {
+  test('when keys is not an array it passes with warns', () => {
     // eslint-disable-next-line no-console
```
Comment (haldarmahesh) [inline src/__test__/index.test.js]: console error is disabled, hence lets remove all these comments as well.

File: src/__test__/index.test.js [hunk 4]
```diff
@@ -6,33 +6,36 @@ import { render, flushEffects, cleanup, fireEvent } from 'react-testing-library'
...
     console.warn = jest.fn();
     render(<TestComponent callback={jest.fn()} detectKeys="someKey" />);
     // eslint-disable-next-line no-console
     expect(console.warn).toHaveBeenCalledWith('Keys should be array!');
   });
+
+  test('when the passed keyevent is an invalid event, an error is thrown', () => {
+    // eslint-disable-next-line no-console
```
Comment (haldarmahesh) [inline src/__test__/index.test.js]: same as above

**Comment by haldarmahesh** [comment]
@gurschitz looks great. I have added some minor comments, please check and let me know what you think.

Thanks for adding `react-testing-library` how can see testing this is much easier now. 
Cheers 🎉

File: src/index.js (hunk 1 above)
Comment (gurschitz) [reply src/index.js]: Ah yes! That makes sense, definitely a good idea. Can you change this when merging?

File: src/index.js (hunk 2 above)
Comment (gurschitz) [reply src/index.js]: Ok!

**Comment by haldarmahesh** [comment]
Thanks, @gurschitz for raising PR.
Cheers 🍻

**Pull request merged**